packages/coin-support-evm/src/operations/prepareTransaction/index.ts

```diff
@@ -46,38 +46,41 @@ export const prepareTransaction = async (
   );
 
   const outputsAddresses = validateAddresses(params, coin);
-  const gasLimit =
-    txn.userInputs.gasLimit ??
-    (await estimateGasLimit(coin.id, {
-      from: account.xpubOrAddress,
-      to:
-        txn.userInputs.outputs[0].address && outputsAddresses[0]
-          ? txn.userInputs.outputs[0].address
-          : account.xpubOrAddress,
-      value: '0',
-      data: '0x',
-    }));
+  const gasLimitEstimate = await estimateGasLimit(coin.id, {
+    from: account.xpubOrAddress,
+    to:
+      txn.userInputs.outputs[0].address && outputsAddresses[0]
+        ? txn.userInputs.outputs[0].address
+        : account.xpubOrAddress,
+    value: '0',
+    data: '0x',
+  });
+  const gasLimit = txn.userInputs.gasLimit ?? gasLimitEstimate;
+  const output = { ...txn.userInputs.outputs[0] };
   const gasPrice = txn.userInputs.gasPrice ?? txn.staticData.averageGasPrice;
   const fee = new BigNumber(gasLimit).multipliedBy(gasPrice);
   const hasEnoughBalance = new BigNumber(account.balance)
-    .minus(txn.userInputs.outputs[0].amount)
+    .minus(output.amount || '0')
     .minus(fee)
     .isPositive();
 
+  if (txn.userInputs.isSendAll && hasEnoughBalance) {
+    output.amount = new BigNumber(account.balance).minus(fee).toString(10);
+    // update userInput so that the max amount is editable & not reset to 0
+    txn.userInputs.outputs[0].amount = output.amount;
+  }
+  txn.userInputs.gasLimit = gasLimit;
```

Comment:
Logic looks incorrect for sendAll when amount is specificed

Reply:
good catch. thanks. updated